without this patch gtk-fixxref must be invoked with '--module' option
and it's bad (f.e. building evolution-exchange-2.28.2 fails)
diff -up gtk-doc-1.13/gtkdoc-fixxref.in.fix-fixxref gtk-doc-1.13/gtkdoc-fixxref.in
--- gtk-doc-1.13/gtkdoc-fixxref.in.fix-fixxref	2009-12-17 14:36:51.000000000 -0500
+++ gtk-doc-1.13/gtkdoc-fixxref.in	2010-01-06 17:56:55.457121567 -0500
@@ -168,36 +168,38 @@ foreach my $dir (@EXTRA_DIRS) {
     }
 }
 
-open (INPUT, "$MODULE-sections.txt")
-	|| die "Can't open $MODULE-sections.txt: $!";
-my $subsection = "";
-while (<INPUT>) {
-    if (m/^#/) {
-        next;
-
-    } elsif (m/^<SECTION>/) {
-        $subsection = "";
-    } elsif (m/^<SUBSECTION\s*(.*)>/i) {
-        $subsection = $1;
-    } elsif (m/^<SUBSECTION>/) {
-        next;
-    } elsif (m/^<TITLE>(.*)<\/TITLE>/) {
-        next;
-    } elsif (m/^<FILE>(.*)<\/FILE>/) {
-        next;
-    } elsif (m/^<INCLUDE>(.*)<\/INCLUDE>/) {
-        next;
-    } elsif (m/^<\/SECTION>/) {
-        next;
-    } elsif (m/^(\S+)/) {
-        my $symbol=CreateValidSGMLID($1);
-
-        if ($subsection eq "Standard" || $subsection eq "Private") {
-            $NoLinks{$symbol} = 1;
+if (defined($MODULE)) {
+    open (INPUT, "$MODULE-sections.txt")
+            || die "Can't open $MODULE-sections.txt: $!";
+    my $subsection = "";
+    while (<INPUT>) {
+        if (m/^#/) {
+            next;
+
+        } elsif (m/^<SECTION>/) {
+            $subsection = "";
+        } elsif (m/^<SUBSECTION\s*(.*)>/i) {
+            $subsection = $1;
+        } elsif (m/^<SUBSECTION>/) {
+            next;
+        } elsif (m/^<TITLE>(.*)<\/TITLE>/) {
+            next;
+        } elsif (m/^<FILE>(.*)<\/FILE>/) {
+            next;
+        } elsif (m/^<INCLUDE>(.*)<\/INCLUDE>/) {
+            next;
+        } elsif (m/^<\/SECTION>/) {
+            next;
+        } elsif (m/^(\S+)/) {
+            my $symbol=CreateValidSGMLID($1);
+
+            if ($subsection eq "Standard" || $subsection eq "Private") {
+                $NoLinks{$symbol} = 1;
+            }
         }
     }
+    close (INPUT);
 }
-close (INPUT);
 
 &FixCrossReferences ($MODULE_DIR);
 
